dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,580 @@
+        happens with ``metadata_``. These dictionaries both have keys
+
+        * ``bracket``, an int representing how strongly the
+          SuccessiveHalvingSearchCV class adapts to history.
+          ``bracket == 0`` stops estimators the least frequently.
+        * ``estimators``, an int representing how many models will be/is created.
+        * ``partial_fit_calls``, an int representing how many times
+           ``partial_fit`` will be/is called.
+        * ``SuccessiveHalvingSearchCV params``, a dictionary representing the
+          parameters used to create the different brackets.
+        * ``decisions``, the number of times ``partial_fit`` will be/is called on
+          each model
+
+        These dictionaries are the same if ``patience`` is not specified. If
+        it is specified, ``partial_fit_calls`` in ``metadata_`` will likely
+        be lower and correspondingly,
+        ``decisions`` in ``metadata_`` will be longer.
+
+    cv_results_ : Dict[str, np.ndarray]
+        This dictionary has keys
+
+        * ``mean_partial_fit_time``
+        * ``mean_score_time``
+        * ``std_partial_fit_time``
+        * ``std_score_time``
+        * ``test_score``
+        * ``rank_test_score``
+        * ``model_id``
+        * ``partial_fit_calls``
+        * ``params``
+        * ``param_{key}``, where ``key`` is every key in ``params``.
+
+        The values in the ``test_score`` key correspond to the last score a model
+        received on the hold out dataset. The key ``model_id`` corresponds with
+        ``history_``. This dictionary can be imported into Pandas.
+
+        In the ``model_id``, the bracket ID prefix corresponds to how strongly
+        that bracket adapts to history. i.e., ``bracket=0`` corresponds to a
+        completely passive bracket that doesn't adapt at all.
+
+    model_history_ : dict of lists of dict
+        A dictionary of each models history. This is a reorganization of
+        ``history_``: the same information is present but organized per model.
+
+        This data has the structure  ``{model_id: hist}`` where ``hist`` is a
+        subset of ``history_`` and ``model_id`` are model identifiers.
+
+    history_ : list of dicts
+        Information about each model after each ``partial_fit`` call. Each dict
+        the keys
+
+        * ``partial_fit_time``
+        * ``score_time``
+        * ``score``
+        * ``model_id``
+        * ``params``
+        * ``partial_fit_calls``
+
+        The key ``model_id`` corresponds to the ``model_id`` in ``cv_results_``.
+        This list of dicts can be imported into Pandas.
+
+    best_estimator_ : BaseEstimator
+        The model with the highest validation score among all the models
+        retained by the "inverse decay" algorithm.
+
+    best_score_ : float
+        Score achieved by ``best_estimator_`` on the vaidation set after the
+        final call to ``partial_fit``.
```

Comment:
Should this docstring use "validation" or "test"?

Reply:
Or "hold-out"?

Reply:
I added a note in the docs for `test_size`:

    test_size : float
        Fraction of the dataset to hold out for computing test/validation
        scores. Defaults to the size of a single partition of
        the input training set.

This is referred to in other places:

    best_estimator_ : BaseEstimator
        The model with the highest validation score as selected by
        the Hyperband model selection algorithm.

I think "test/validation" is succinct definition for "estimators are scored on data they've never seen before".